sdcm/nemesis.py

```diff
@@ -2098,23 +2098,19 @@ def choose_partitions_for_delete(self, partitions_amount, ks_cf, with_clustering
 
         # In the large partition tests part of partitions are used for data validation. We do not want to delete a data in those partitions
         # to prevent scylla-bench command failure
-        max_partitions_in_test_table = self.cluster.params.get('max_partitions_in_test_table')
-        partition_range_with_data_validation = self.cluster.params.get('partition_range_with_data_validation')
-
-        if partition_range_with_data_validation and '-' in partition_range_with_data_validation:
-            partition_range_splitted = partition_range_with_data_validation.split('-')
-            start_range = int(partition_range_splitted[0])
-            end_range = int(partition_range_splitted[1])
-            exclude_partitions.extend(i for i in range(start_range, end_range))
-            if partitions_amount == max_partitions_in_test_table:
-                partitions_amount -= end_range - start_range
+
+        if partitions_attrs := self.tester.partitions_attrs:
+            if partitions_attrs.partition_range_with_data_validation:
+                exclude_partitions.extend(
+                    i for i in range(partitions_attrs.partition_start_range, partitions_attrs.partition_end_range))
+                partitions_amount = min(partitions_amount, partitions_attrs.non_validated_partitions)
         self.log.debug(f"Partitions amount for delete : {partitions_amount}")
 
         partitions_for_delete = defaultdict(list)
         with self.cluster.cql_connection_patient(self.target_node, connect_timeout=300) as session:
             session.default_consistency_level = ConsistencyLevel.ONE
 
-            for partition_key in [i * 2 + 50 for i in range(max_partitions_in_test_table)]:
+            for partition_key in [i * 2 + 50 for i in range(self.tester.partitions_attrs.max_partitions_in_test_table)]:
```

Comment:
I know it's not new in this PR but can you explain why we do the loop from "i * 2 + 50"?

Reply:
@roydahan , right, this code probably badly works. it is rewritten now like:
```
-            for partition_key in [i * 2 + 50 for i in range(self.tester.partitions_attrs.max_partitions_in_test_table)]:
-                if len(partitions_for_delete) == partitions_amount:
-                    break
+            while available_partitions_for_deletion and len(partitions_for_delete) < partitions_amount:
+                random_index = random.randint(0, len(available_partitions_for_deletion) - 1)
+                partition_key = available_partitions_for_deletion.pop(random_index)
```

Reply:
@juliayakovlev , following this PR refactor, i changed this code, please look and comment anything for that.